homeassistant/components/fritzbox/__init__.py

```diff
@@ -84,8 +106,36 @@ async def async_setup_entry(hass, entry):
     except LoginError as err:
         raise ConfigEntryAuthFailed from err
 
-    hass.data.setdefault(DOMAIN, {CONF_CONNECTIONS: {}, CONF_DEVICES: set()})
-    hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id] = fritz
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        CONF_CONNECTIONS: fritz,
+    }
+
+    async def async_update_coordinator():
+        """Fetch all device data."""
+        data = {}
+        for device in await hass.async_add_executor_job(fritz.get_devices):
+            device: FritzhomeDevice
+            try:
+                await hass.async_add_executor_job(device.update)
+            except requests.exceptions.HTTPError as ex:
+                fritz.login()
```

Comment:
Should this retry after login instead of raising, then raise `ConfigEntryAuthFailed` if it really can't login?

Reply:
to be honest I simple overtake this logic from previous entity's `update()` methods :see_no_evil: 
`ConfigEntryAuthFailed` is already handled some lines above.
I assume the re-login during `update()` was intended in case the Fritzbox has rebooted and therefore a re-login was needed :thinking:

Reply:
If the password gets changed, and auth start failing, and you raise `ConfigEntryAuthFailed` during the update, the coordinator will stop updates and start a reauth flow.

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L215